fintrack/src/components/UserSignIn.js

```diff
@@ -0,0 +1,69 @@
+import { useContext, useRef, useState } from 'react'
+import { Link, useNavigate, useLocation } from 'react-router-dom'
+import ErrorsDisplay from './ErrorsDisplay'
+import UserContext from '../context/UserContext'
+
+const UserSignIn = () => {
+
+  const { actions } = useContext(UserContext)
+  const navigate = useNavigate()
+  const location = useLocation() // contains the state object passed from the PrivateRoute component
+  console.log(location)
+
+  // State
+  const email = useRef(null)
+  const password = useRef(null)
```

Comment:
Why did you choose to use refs here, and would would be the alternative to using refs in react? What would be the pros and cons of each choice?